src/backend/joanie/core/models/products.py

```diff
@@ -804,12 +849,26 @@ def clean(self):
                     error_dict["order_group"].append(
                         f"An active order group is required for product {product_title:s}."
                     )
-                else:
-                    nb_seats = self.order_group.nb_seats
-                    if 0 < nb_seats <= self.order_group.get_nb_binding_orders():
+                elif self.order_group.is_enabled:
+                    if self.order_group.nb_seats == 0:
                         error_dict["order_group"].append(
-                            f"Maximum number of orders reached for product {product_title:s}"
+                            f"The order group for product {product_title:s} does not accept "
+                            "anymore order, the number of seat has been set to 0."
                         )
+                    elif self.order_group.nb_seats is not None:
+                        if not self.order_group.available_seats:
+                            error_dict["order_group"].append(
+                                f"Maximum number of orders reached for product {product_title:s}"
+                            )
```

Comment:
IMO We don't care to distinguish the case where `nb_seats` is set to `0`

```suggestion
                elif self.order_group.is_enabled and self.order_group.nb_seats is not None:
                        if self.order_group.available_seats == 0:
                            error_dict["order_group"].append(
                                f"Maximum number of orders reached for product {product_title:s}"
                            )
```